File: TLS Inspector/Supporting Files/ActionTipTarget.m
#import "ActionTipTarget.h"

@implementation ActionTipTarget

+ (ActionTipTarget *) targetWithView:(UIView *)view {
    ActionTipTarget * target = [ActionTipTarget new];
    target.targetView = view;
    return target;
}

+ (ActionTipTarget *) targetWithBarButtonItem:(UIBarButtonItem *)barButtonItem {
    ActionTipTarget * target = [ActionTipTarget new];
    target.targetBarButtonItem = barButtonItem;
    return target;
}

@end

File: tls-inspector/User Interface/ActionTipTarget.swift
import UIKit

class ActionTipTarget {
    private var view: UIView?
    private var barButtonItem: UIBarButtonItem?

    init(view: UIView) {
        self.view = view
    }

    init(barButtonItem: UIBarButtonItem) {
        self.barButtonItem = barButtonItem
    }

    public func attach(to: UIPopoverPresentationController?) {
        if let view = self.view {
            to?.sourceView = view
        } else if let barButtonItem = self.barButtonItem {
            to?.barButtonItem = barButtonItem
        }
    }
}
